cmd/zapi/cmd/post/post.go

```diff
@@ -0,0 +1,125 @@
+package post
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"sync/atomic"
+	"time"
+
+	"github.com/brimsec/zq/cmd/zapi/cmd"
+	"github.com/brimsec/zq/cmd/zapi/format"
+	"github.com/brimsec/zq/pkg/display"
+	"github.com/brimsec/zq/zqd/api"
+	"github.com/mccanne/charm"
+)
+
+var Post = &charm.Spec{
+	Name:  "post",
+	Usage: "post [options] path...",
+	Short: "post log file(s) to a space",
+	New:   NewPost,
+}
+
+func init() {
+	cmd.CLI.Add(Post)
+}
+
+type LogCommand struct {
+	*cmd.Command
+	force      bool
+	bytesRead  int64
+	bytesTotal int64
+	start      time.Time
+	done       bool
+}
+
+func NewPost(parent charm.Command, flags *flag.FlagSet) (charm.Command, error) {
+	c := &LogCommand{Command: parent.(*cmd.Command)}
+	flags.BoolVar(&c.force, "f", false, "create space if specified space does not exist")
+	return c, nil
+}
+
+// Run lists all spaces in the current zqd host or if a parameter
+// is provided (in glob style) lists the info about that space.
+func (c *LogCommand) Run(args []string) (err error) {
+	client := c.Client()
+	if len(args) == 0 {
+		return errors.New("path arg(s) required")
+	}
+	if c.force {
+		sp, err := client.SpacePost(c.Context(), api.SpacePostRequest{Name: c.Spacename})
+		if err != nil && err != api.ErrSpaceExists {
+			return err
+		}
+		c.Spacename = sp.Name
+	}
+	var out io.Writer
+	var dp *display.Display
+	if !c.NoFancy {
+		dp = display.New(c, time.Second)
+		out = dp.Bypass()
+		go dp.Run()
+	} else {
+		out = os.Stdout
+	}
+	id, err := c.SpaceID()
+	if err != nil {
+		return err
+	}
+	c.start = time.Now()
+	stream, err := client.LogPost(c.Context(), id, api.LogPostRequest{Paths: args})
+	if err != nil {
+		return err
+	}
+	var payload interface{}
+loop:
+	for {
+		payload, err = stream.Next()
+		if err != nil {
+			break loop
+		}
+		if payload == nil {
+			break loop
+		}
+		switch typ := payload.(type) {
+		case *api.LogPostWarning:
+			fmt.Fprintf(out, "warning: %s\n", typ.Warning)
+		case *api.TaskEnd:
+			if typ.Error != nil {
+				err = typ.Error
+			}
+			break loop
+		case *api.LogPostStatus:
+			atomic.StoreInt64(&c.bytesRead, typ.LogReadSize)
+			atomic.StoreInt64(&c.bytesTotal, typ.LogTotalSize)
+		}
+	}
```

Comment:
```suggestion
	for {
		var v interface{}
		v, err = stream.Next()
		if err != nil || v == nil {
			break
		}
		switch v := v.(type) {
		case *api.LogPostWarning:
			fmt.Fprintf(out, "warning: %s\n", v.Warning)
		case *api.TaskEnd:
			err = v.Error
			break
		case *api.LogPostStatus:
			atomic.StoreInt64(&c.bytesRead, v.LogReadSize)
			atomic.StoreInt64(&c.bytesTotal, v.LogTotalSize)
		}
	}
```

Reply:
The break in your switch statement won't exit the for loop and err will get replaced with whatever error is in stream.Next().

Reply:
🤦 once more.